rllib/utils/actor_manager.py

```diff
@@ -0,0 +1,324 @@
+from collections import defaultdict
+import logging
+from typing import Callable, List, Optional, Tuple, TypeVar
+
+import ray
+from ray.actor import ActorHandle
+from ray.exceptions import RayActorError
+from ray.util.annotations import DeveloperAPI
+
+
+logger = logging.getLogger(__name__)
+# Generic type var for foreach_* methods.
+T = TypeVar("T")
+
+
+class FaultTolerantActorManager:
+    """A manager that is aware of the healthiness states of remote actors.
```

Comment:
Can you include the examples and their expected behavior for at least the use-cases that you are unittesting? After reading the unittests I am not sure if I can confidently explain what the expected behavior of this class is :) so some documentation here is indeed needed.